component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -517,5 +529,54 @@ private void addMatchStringsToList(Matcher matcher, Set<String> matches) {
+            UserStoreManager um = X509CertificateUtil.getUserRealm(userIdentifier).getUserStoreManager();
+            String[] filteredUsers = um.listUsers(MultitenantUtils.getTenantAwareUsername(userIdentifier),
+                    X509CertificateConstants.MAX_ITEM_LIMIT_UNLIMITED);
+            if (filteredUsers.length == 1) {
+                if (log.isDebugEnabled()) {
+                    log.debug("User exists with the user name: " + userIdentifier);
+                }
+                return getUserStoreDomainNameHelper(filteredUsers[0]);
+            } else if (filteredUsers.length > 1) {
+                authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                        X509CertificateConstants.USERNAME_CONFLICT);
+                throw new AuthenticationFailedException("Conflicting users with user name: " + userIdentifier);
+            } else if (getAuthenticatorConfig().getParameterMap().containsKey(X509CertificateConstants.LOGIN_CLAIM_URIS)) {
+                String[] multiAttributeClaimUris = getAuthenticatorConfig().getParameterMap()
+                        .get(X509CertificateConstants.LOGIN_CLAIM_URIS).split(",");
+                for (String multiAttributeClaimUri : multiAttributeClaimUris) {
+                    String[] usersWithClaim = ((AbstractUserStoreManager) X509CertificateUtil.getUserRealm(userIdentifier)
+                            .getUserStoreManager()).getUserList(multiAttributeClaimUri, userIdentifier, null);
+                    if (usersWithClaim.length == 1) {
+                        return getUserStoreDomainNameHelper(usersWithClaim[0]);
+                    } else if (usersWithClaim.length > 1) {
+                        authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                                X509CertificateConstants.USERNAME_CONFLICT);
+                        throw new AuthenticationFailedException("Conflicting users with claim value: " + userIdentifier);
+                    }
+                }
+                throw new AuthenticationFailedException("Unable to find X509 Certificate's user in user store. ");
+            } else {
+                authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                        X509CertificateConstants.USER_NOT_FOUND);
+                throw new AuthenticationFailedException("Unable to find X509 Certificate's user in user store. ");
+            }
+        } else {
+            return X509CertificateConstants.PRIMARY;
+        }
+    }
+
+    private String getUserStoreDomainNameHelper(String user) {
+
+        if (user.indexOf("/") > 0) {
+            String[] subjectIdentifierSplits = user.split("/", 2);
+            return subjectIdentifierSplits[0];
+        } else {
+            return X509CertificateConstants.PRIMARY;
```

Comment:
No need to define a new constant. We can reuse the constant in [1].
[1] https://github.com/wso2/carbon-kernel/blob/4.5.x/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/UserCoreConstants.java#L65